datamade/just-spaces
# Initialize frontend app with Parcel and ES6

## Overview

Initialize the Just Spaces frontend as a separate Django app. In the frontend app, scaffold a JavaScript dev environment with support for the following features:

- ES6 syntax with Babel
- Module bundling with Parcel
- Linting with JSHint
- Tests with Jasmine

### Notes

There's a lot going on here, so I'll leave comments on the diff.

## Testing Instructions

- Make sure you have Docker installed ([MacOS link](https://docs.docker.com/docker-for-mac/install/))
- Install dev Python dependencies with `pip install -U -r dev-requirements.txt`
- Run `docker-compose build` to build the NPM image
- In one shell, run `docker-compose up` to bundle and watch the frontend assets
- In another shell, run `python manage.py runserver` to run the dev server
- Navigate to http://localhost:8000, open the console, and confirm that you see the "Hello, World!" message
- Run Jasmine tests with `docker-compose run --rm frontend test`
- Run JSHint linting with `docker-compose run --rm frontend lint`

Connects #55, https://github.com/datamade/ops/issues/593
